src/libtrx/game/game_flow/reader_tr2.def.c

```diff
@@ -124,6 +124,20 @@ static void M_LoadRoot(JSON_OBJECT *const obj, GAME_FLOW *const gf)
     gf->settings = m_DefaultSettings;
     M_LoadSettings(obj, &gf->settings);
 
+    const char *tmp_s =
+        JSON_ObjectGetString(obj, "main_menu_picture", JSON_INVALID_STRING);
+    if (tmp_s == JSON_INVALID_STRING) {
+        Shell_ExitSystem("'main_menu_picture' must be a string");
+    }
+    gf->main_menu_background_path = Memory_DupStr(tmp_s);
+
+    tmp_s =
+        JSON_ObjectGetString(obj, "savegame_fmt_legacy", JSON_INVALID_STRING);
+    if (tmp_s == JSON_INVALID_STRING) {
+        Shell_ExitSystem("'savegame_fmt_legacy' must be a string");
+    }
+    gf->savegame_fmt_legacy = Memory_DupStr(tmp_s);
```

Comment:
Can this be consolidated in the common file? I think we're now duplicating this block between both games.

Reply:
I can combine the functions overall, it results in +78,-99 diff. Or do you just mean the picture and format parts, putting them into a new common function?

Reply:
Just the picture and the file format properties that in my understanding are now shared.

Reply:
Thanks, done.